tensorflow/core/grappler/optimizers/remapper.cc

```diff
@@ -705,6 +705,23 @@ bool IsBiasSemanticAdd(const RemapperContext& ctx,
   return false;
 }
 
+void AddInputShapesAttr(const RemapperContext& ctx, int node_index) {
```

Comment:
What if some dimensions are unknown like -1, -2 etc.

Reply:
Hi @mdfaijul, in order to infer dimensions of the operation we are using method `InferStatically` from `graph_properties`. If it cannot infer dimensions or there are unknown then method `GetInputProperties()` would return empty array and we would not add `_input_shapes` attribute to nodes and as a result we would revert to default behaviour and not use dimensions to decide whether we want to rewrite not.